src/libraries/Common/src/Interop/Browser/Interop.Runtime.cs

```diff
@@ -0,0 +1,509 @@
+            }
+        }
+
+        internal static object CreateTaskSource(int js_id)
+        {
+            return new TaskCompletionSource<object>();
+        }
+
+        internal static void SetTaskSourceResult(TaskCompletionSource<object> tcs, object result)
+        {
+            tcs.SetResult(result);
+        }
+
+        internal static void SetTaskSourceFailure(TaskCompletionSource<object> tcs, string reason)
+        {
+            tcs.SetException(new JSException(reason));
+        }
+
+        internal static int GetTaskAndBind(TaskCompletionSource<object> tcs, int js_id)
+        {
+            return BindExistingObject(tcs.Task, js_id);
+        }
+
+        internal static int BindExistingObject(object raw_obj, int js_id)
+        {
+            JSObject? obj = raw_obj as JSObject;
+
+            if (obj == null && !raw_to_js.TryGetValue(raw_obj, out obj))
+                raw_to_js[raw_obj] = obj = new JSObject(js_id, raw_obj);
+
+            return obj == null ? 0 : (int)(IntPtr)obj.Handle;
+        }
+
+        internal static int GetJSObjectId(object raw_obj)
+        {
+            JSObject? obj = raw_obj as JSObject;
+
+            if (obj == null && !raw_to_js.TryGetValue(raw_obj, out obj))
+                return -1;
+
+            return obj != null ? obj.JSHandle : -1;
+        }
+
+        internal static object? GetMonoObject(int gc_handle)
+        {
+            GCHandle h = (GCHandle)(IntPtr)gc_handle;
+            JSObject? o = h.Target as JSObject;
+            if (o != null && o.RawObject != null)
+                return o.RawObject;
+            return o;
+        }
+
+        internal static object BoxInt(int i)
+        {
+            return i;
+        }
+        internal static object BoxDouble(double d)
+        {
+            return d;
+        }
+
+        internal static object BoxBool(int b)
+        {
+            return b == 0 ? false : true;
+        }
+
+        internal static bool IsSimpleArray(object a)
+        {
+            if (a is Array arr)
+            {
+                if (arr.Rank == 1 && arr.GetLowerBound(0) == 0)
+                    return true;
+            }
+            return false;
+
+        }
+
+        internal static object? GetCoreType(string coreObj)
+        {
+            Assembly asm = typeof(Runtime).Assembly;
+            Type? type = asm.GetType(coreObj);
+            return type;
+
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        internal struct IntPtrAndHandle
+        {
+            [FieldOffset(0)]
+            internal IntPtr ptr;
+
+            [FieldOffset(0)]
+            internal RuntimeMethodHandle handle;
+        }
+
+        internal static string GetCallSignature(IntPtr method_handle)
+        {
+            IntPtrAndHandle tmp = default(IntPtrAndHandle);
+            tmp.ptr = method_handle;
+
+            var mb = MethodBase.GetMethodFromHandle(tmp.handle);
+
+            string res = "";
```

Comment:
Using string concatenation is not efficient here. You could use char[] array instead and then return new string(array)

Reply:
addressed in recent commits